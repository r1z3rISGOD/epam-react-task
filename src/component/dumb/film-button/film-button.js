"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilmButton = void 0;
var React = require("react");
var react_redux_1 = require("react-redux");
var react_router_dom_1 = require("react-router-dom");
var actions_1 = require("../../../store/actions");
var styled_components_1 = require("styled-components");
var Button = styled_components_1.default.button(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  width: 130px;\n  height: 50px;\n  font-size: 1rem;\n  font-family: \"Roboto\", sans-serif;\n  color: #e55f64;\n  font-weight: bold;"], ["\n  width: 130px;\n  height: 50px;\n  font-size: 1rem;\n  font-family: \"Roboto\", sans-serif;\n  color: #e55f64;\n  font-weight: bold;"])));
var FilmButton = function () {
    var dispatch = react_redux_1.useDispatch();
    var history = react_router_dom_1.useHistory();
    return (React.createElement(Button, { onClick: function () {
            history.push('/');
            dispatch(actions_1.clearStore());
            console.clear();
        } }, "SEARCH"));
};
exports.FilmButton = FilmButton;
var templateObject_1;
//# sourceMappingURL=film-button.js.map