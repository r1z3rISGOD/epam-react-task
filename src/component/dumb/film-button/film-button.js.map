{"version":3,"file":"film-button.js","sourceRoot":"","sources":["film-button.tsx"],"names":[],"mappings":";;;;;;;AAAA,6BAA8B;AAC9B,uDAAsC;AAEtC,IAAM,MAAM,GAAG,2BAAM,CAAC,MAAM,4MAAA,yIAMP,IAAA,CAAA;AAMd,IAAM,UAAU,GAA+B,UAAC,EAAW;QAAT,OAAO,aAAA;IAC9D,OAAO,CACD,oBAAC,MAAM,IAAC,SAAS,EAAC,KAAK,EAAC,OAAO,EAAE,cAAM,OAAA,OAAO,EAAE,EAAT,CAAS,aAAiB,CACtE,CAAA;AACH,CAAC,CAAA;AAJY,QAAA,UAAU,cAItB"}