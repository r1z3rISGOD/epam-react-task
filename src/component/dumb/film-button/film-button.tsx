import * as React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  width: 130px;
  height: 50px;
  font-size: 1rem;
  font-family: "Roboto", sans-serif;
  color: #e55f64;
  font-weight: bold;`

interface FilmButtonProps {
    onClick: any
}

export const FilmButton : React.FC<FilmButtonProps> = ({ onClick }) => {
  return (
        <Button className='btn' onClick={() => onClick()}>SEARCH</Button>
  )
}
