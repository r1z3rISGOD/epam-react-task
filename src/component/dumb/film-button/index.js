import { FilmButton } from './film-button'
export { FilmButton }
