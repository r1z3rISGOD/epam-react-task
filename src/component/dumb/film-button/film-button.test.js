import React from 'react'
import { FilmButton } from './film-button.tsx'

const mockOnClick = jest.fn(() => 1)

const setUp = () => shallow(<FilmButton onClick={mockOnClick}/>)

it('film button', () => {
  const comp = setUp()
  expect(comp).toMatchSnapshot()
  const btn = comp.find('.btn')
  btn.simulate('click')
  expect(mockOnClick.mock.calls.length).toBe(1)
})
