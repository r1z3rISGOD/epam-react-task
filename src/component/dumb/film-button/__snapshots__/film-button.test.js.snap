// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`film button 1`] = `
<styled.button
  className="btn"
  onClick={[Function]}
>
  SEARCH
</styled.button>
`;
